device_doctor/lib/src/health.dart

```diff
@@ -82,21 +82,18 @@ class HealthCheckResult {
 }
 
 /// Check healthiness for discovered devices.
-///
-/// If any failed check, an explanation message will be sent to stdout and
-/// an exception will be thrown.
-Future<void> healthcheck(Map<String, List<HealthCheckResult>> deviceChecks) async {
+Future<Map<String, String>> healthcheck(Map<String, List<HealthCheckResult>> deviceChecks) async {
+  final Map<String, String> healthcheckMap = <String, String>{};
   if (deviceChecks.isEmpty) {
-    stderr.writeln('No healthy device is available');
+    healthcheckMap['Attached device'] = 'No device is available';
```

Comment:
can you make the key and value be constants here too, like you did for other keys. And the key be a little more machine friendly, like:

attached_device

?